### ISSUE TITLE
All textboxes are missing options to formate inserted text

### ISSUE THREAD

**Issue by dagraf** [opencast-ilias/OpenCast#50]
**Describe the bug**
All textboxes are missing options to formate inserted text (i.e. "Work instruction" in Settings-Tab of a series, "Info Text" in sub-tab "Events" of the plugins configuration possibility

**To Reproduce**
1. Go to a series an open the tab 'Settings'
2. See missing options to format the written text

**Expected behavior**
Written text should be formatable (bold, italic, ..., hyperlink, source code).

**Screenshots**
v3.6.1:
<img width="624" alt="Bildschirmfoto 2022-06-20 um 20 20 10" src="https://user-images.githubusercontent.com/15246287/174659813-623faa8d-c5e3-472e-af90-2fd99c154155.png">
<img width="788" alt="Bildschirmfoto 2022-06-20 um 20 24 01" src="https://user-images.githubusercontent.com/15246287/174659911-d61a270e-16a1-4944-9d28-72fb492fa0c8.png">

v4.0.1
<img width="1121" alt="Bildschirmfoto 2022-06-20 um 20 23 36" src="https://user-images.githubusercontent.com/15246287/174659898-e28bc20b-440d-4538-8031-772604c898b6.png">
<img width="1180" alt="Bildschirmfoto 2022-06-20 um 20 23 47" src="https://user-images.githubusercontent.com/15246287/174659925-cfa6128a-b1e1-44c5-9a8a-f12b7d2b96d9.png">

**Comment by okaufman** [comment]
The old tinyMCE-Editor can no longer be used in the new ILIAS-Forms. The Editor will be deprecated in more and more places in ILIAS. Soon the tinyMCE-Editor wil be replaced by a markdown editor. But the markdown editor is not yet available.

We could parse the output for possible links and make them clickable. I am assuming that the editor is mostly used to insert links.

**Comment by dagraf** [comment]
Ok, let's implement "parsing the output for possible links and make them clickable" until the new markdown editor is implemented. @okaufman Could you please do this and give us an update when the markdown editor is working? Thx!

**Comment by chfsx** [comment]
The Markdown-Input has been implemented for ILIAS 9: https://github.com/ILIAS-eLearning/ILIAS/pull/5152

**Comment by dagraf** [comment]
> The Markdown-Input has been implemented for ILIAS 9: [ILIAS-eLearning/ILIAS#5152](https://github.com/ILIAS-eLearning/ILIAS/pull/5152)

@chfsx > Which means that this issue will be solved with ILIAS 9?

**Comment by chfsx** [comment]
@dagraf This does not directly solve the problem, we will then also include this form field (type) in the plugin so that it can be used. We also have to "migrate" the existing content in some way. So far, HTML was generated and saved with the old input, with the new input we will generate and save Markdown. Therefore, the existing content would have to be converted once from HTML to Markdown, which will not work in all cases. or we remove all HTML tags once.

**Comment by dagraf** [comment]
@chfsx Thx for the explanation. About migration we start to think when ILIAS 9 is around the corner. One question I still have: Does this new form field include visual UI options (meaning buttons) to format text?

**Comment by chfsx** [comment]
@dagraf yes there are several buttons which help for the most common actions:

![image](https://github.com/opencast-ilias/OpenCast/assets/6661332/6a343c0f-60a7-4562-891f-1553e30d6ccd)

**Comment by dagraf** [comment]
> @dagraf This does not directly solve the problem, we will then also include this form field (type) in the plugin so that it can be used. We also have to "migrate" the existing content in some way. So far, HTML was generated and saved with the old input, with the new input we will generate and save Markdown. Therefore, the existing content would have to be converted once from HTML to Markdown, which will not work in all cases. or we remove all HTML tags once.

@chfsx Did you include this form field (type) describe here when you created the PR for ILIAS 9?

**Comment by chfsx** [comment]
@dagraf currently not, every field that used the Tiny before, must be migrated to the new Markdown Input. But the content which users created with the old Tiny-Editor, should be migrated as well. the problem behind this: the Tiny generated HTML-Code, the Markdown only supports Markdown. There is no reliable way to migrate HTML to Markdown. I think we must discuss what happens to the already stored HTML code in those cases.

**Comment by dagraf** [comment]
Ok. Lets discuss this in the next community meeting. Because the format options have been gone since the summer of 2022, I would suggest not investing too much effort here. Since I assume, that only a few lecturers did use those options.

**Comment by dagraf** [comment]
In meeting:
- Form is now definitively there in ILIAS 9. But if we want to use it we need to refactor the whole form.
- Problem: Already existing formated inputs can not be migrated from html to markdown. Mainly the editing of migrated inputs will be problematic
- Decision: We concentrate on migrating the pure content, not the existing formated content. But this can wait. I therefore move the issue back to the backlog.
